src/chk/chk_iv.c

```diff
@@ -192,16 +165,26 @@ chk_iv_update(void *ns, struct chk_iv *iv, uint32_t shortcut, uint32_t sync_mode
 	int			rc;
 
 	iv->ci_rank = dss_self_rank();
-	iov.iov_buf = iv;
-	iov.iov_len = sizeof(*iv);
-	iov.iov_buf_len = sizeof(*iv);
-	sgl.sg_nr = 1;
-	sgl.sg_nr_out = 0;
-	sgl.sg_iovs = &iov;
-
-	memset(&key, 0, sizeof(key));
-	key.class_id = IV_CHK;
-	rc = ds_iv_update(ns, &key, &sgl, shortcut, sync_mode, 0, retry);
+
+	if (chk_is_on_leader(iv->ci_gen, -1, false) && iv->ci_to_leader) {
+		/*
+		 * XXX: It is the check engine sends IV message to the check leader on
+		 *	the same rank. Then directly notify the check leader without RPC.
+		 */
+		rc = chk_leader_notify(iv->ci_gen, iv->ci_rank, iv->ci_phase, iv->ci_status);
```

Comment:
this is a bit strange, maybe do this outside of chk_iv_update(), not through iv interface?

Reply:
That is possible, but will cause more condition check points from the the callers. May rename chk_iv_update() can avoid the confusing.